app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt

```diff
@@ -75,6 +77,9 @@ class StateFragmentPresenter @Inject constructor(
   private val routeToHintsAndSolutionListener = activity as RouteToHintsAndSolutionListener
   private val hasConversationView = true
 
+  private var hasExplorationJustLoaded: Boolean = false
```

Comment:
This variable is initialized as follows  ```hasExplorationJustLoaded = savedInstanceState == null```

my original intention for having this variable was that it will only be true if the exploration has just loaded and will be false after that. so we can have something like this in processEphemeralState method

```

if (hasExplorationJustLoaded) 
    recyclerViewAssembler.restoreHintHandlerWithCheckpoint(ephemeralState.explorationCheckpoint)
```

This should have ensured that we only restore the hint handler with the checkpoint once when the exploration is loaded for the first time after it is resumed, and then the hint handler will restore again from the checkpoint only if the checkpoint is saved in the saved instance state. 

But later I found out that this will be false until the fragment is recreated because it is only then that we will reinitialize this variable.  So now I am not really sure that even if we save the checkpoint in the saved instance state how will we restore the exploration for the first time without having it restore the hint handler every time a new checkpoint is available.

Reply:
If we are introducing a new ExplorationProgressController method for tracking whether the hint/solution can be shown (rather than just whether it was shown), then I don't think you need to bother with saved instance state at all since we can just rely on the EphemeralState LiveData to restore the entirety of the StateFragment state (the only thing today we don't get from EphemeralState is whether a hint can be shown which is why Farees needed to work on that PR in the first place; the controller & EphemeralState change suggested above completely mitigates the need for that fix).

Reply:
resolving this because changed the approach to handle hints